src/kudu/consensus/consensus_peers.cc

```diff
@@ -503,8 +503,10 @@ Peer::~Peer() {
 
 RpcPeerProxy::RpcPeerProxy(gscoped_ptr<HostPort> hostport,
                            gscoped_ptr<ConsensusServiceProxy> consensus_proxy)
-    : hostport_(std::move(DCHECK_NOTNULL(hostport))),
-      consensus_proxy_(std::move(DCHECK_NOTNULL(consensus_proxy))) {
+    : hostport_(std::move(hostport)),
```

Comment:
Why does moving the null check inside the body beneficial?

Reply:
DCHECK_NOTNULL does not compile any more. For gscoped ptr, it bombs.
I am trying to not lose DEBUG checks while making the code compile in pure debug mode.